crankshaft-engine/src/lib.rs

```diff
@@ -58,12 +77,18 @@ impl Engine {
             "submitting job{} to the `{}` backend",
             task.name
                 .as_ref()
-                .map(|name| format!(" with name `{}`", name))
+                .map(|name| format!(" with name `{name}`"))
                 .unwrap_or_default(),
             name
         );
 
-        backend.spawn(task, token)
+        let event_sender = if backend.monitored {
+            self.event_sender.clone()
+        } else {
+            None
+        };
+
```

Comment:
I think we can remove these lines in favor of just passing `backend.spawn` a `self.event_sender.clone()` for its second argument.